hw/ip/sram_ctrl/data/sram_ctrl.hjson

```diff
@@ -74,9 +90,12 @@
         { bits: "0"
           name: "ERROR"
           desc: '''
-                Set to 1 if a correctable or uncorrectable error has occurred in the SRAM.
-                If set to 1, SW should check the !!ERROR_TYPE and !!ERROR_ADDRESS register for more information.
-                This bit will be cleared automatically if the errors are cleared in !!ERROR_TYPE.
+                Set to 1 if an uncorrectable parity error has occurred in the SRAM macro.
+                The address of the faulty access is exposed in the !!ERROR_ADDRESS register
+                for debugging purposes. This is an unrecoverable error condition and causes
+                the SRAM controller to immediately discard the scrambling keys and continuously
+                send out sram_integ_alert alerts. Also, the affected SRAM memory will not accept
+                any memory transactions anymore in that case.
```

Comment:
@tjaychen: I made this error case stricter such that the SRAM becomes completely unusable once a parity error has been detected. i.e., bus transactions will be blocked, and scrambling keys will be discarded.

Reply:
If the existing scrambling keys is thrown away, can we still re-arm with a different one?
That way we'll have at least have a mechanism to debug maybe which bit has become faulty (if that is indeed the case).

The alert I think is fine with me, since we can always choose to ignore it or handle it differently on the alert handler end.

Reply:
currently I also block any new TL-UL transactions to memory once a parity failure has been detected - but I can take that countermeasure out such that re-requesting a new key works.

the sram controller also provides a register holding the last faulty address for debugging.

Reply:
wait so let me clarify. What I meant to say was..let's say there is a faulty bit in memory for whatever reason (it's stuck 0 or stuck 1).  What would be the easiest way for someone to debug that without needing to potentially constantly reset? That's what I'm asking really.

Reply:
yeah that is a good point. I looked at this mainly from the countermeasure perspective, but I see that it makes debugging really onerous if TL-UL transactions are going to be blocked.

I have removed this mechanism and just scrap the keys upon parity error now.
that way, you can still interact with the memory after an error for debugging purposes.